### ISSUE TITLE
porting to ROS2, or ROS2 equivalent for the package

### ISSUE THREAD

**Issue by altineller** [NaokiAkai/als_ros#25]
Hello NaokiAkai,

First of all thanks for the hard work. How diffucult would it be to port this package to ROS2? I have gone over the learning curve of ROS2, and I have been able to port few simple things, but I can not estimate how diffucult it would be to undertake to get als ros running on ROS2.

Best Regards,
C.A.

**Comment by SynapseProgramming** [comment]
Hi, having ported packages between ros/ros2, i must say that it is rather tedious/difficult to make the jump from ros1 to ros2. This is mainly due to 1.) conversion from tf to tf2 2.) ros2 rclcpp enforces OOP style of coding (you have to inherit from the node class) 3.) Other dependencies. which may/may not have been ported over. I would probably attempt to port this package sometime in late nov/dec, as my hands are kinda full currently.

**Comment by MFernandezCarmona** [comment]
Hi,
I too would be interested in testing a ROS2 version of your great code. I'll play a bit with it if I can, but please let me know if you make a port of this.
Cheers

**Comment by altineller** [comment]
@SynapseProgramming 

I too, have rolled over the learning curve of ros2, and ported few packages. And this package is a diffucult one. I see no reason why we can't rewrite the ros communications/interface code in ros2, and transfer the core into it.

However, as you know there is also a new navstack on ros2. This package is superior to amcl that comes with ros noetic, but I am not sure if it is better than the default amcl that comes with ros2. If a perforance/robustness gain can be obtained, then I too would like to port this package to ROS2.

Best Regards

**Comment by ajtudela** [comment]
Hi all, 

I'm also interested in the ROS2 port. I forked the repo, created a new branch "ros2" and I started to porting to ROS2 but the code it's hard to follow. So I read the paper.

If I understand it right it, it's a MCL algorithm +  pose tracking + failure detector. So, the best aproach it's expand the [AMCL](https://github.com/ros-planning/navigation2/tree/main/nav2_amcl) algorithm from Nav2. 

- First, adding the class conditional measurement model, [here](https://github.com/ros-planning/navigation2/blob/main/nav2_amcl/include/nav2_amcl/sensors/laser/laser.hpp).
- Then, adding the pose tracking and failure detector.
